src/App/Fossa/IAT/Types.hs

```diff
@@ -21,3 +28,20 @@ data UserDefinedAssertionMeta = UserDefinedAssertionMeta
   , assertedDescription :: Maybe Text
   , assertedUrl :: Maybe Text
   }
+
+instance FromJSON UserDefinedAssertionMeta where
+  parseJSON = withObject "UserDefinedAssertionMetadata" $ \obj -> do
+    UserDefinedAssertionMeta
+      <$> obj .: "name"
+      <*> obj .: "version"
+      <*> obj .: "license"
+      <*> obj .: "description"
+      <*> obj .: "url"
+
+-- | Extract user defined binary dependencies from a list of dependencies.
+-- Evaluates to the list of user defined binaries and the remaining graph.
+extractUserDefinedBinaries :: [Dependency] -> ([Dependency], [Dependency])
```

Comment:
That this exists is a code smell to me; we're still constructing some combined thing ("all of the dependencies") and doing post-processing to split out a subset we care about -- why not keep them separate as part of the generating function?

Further: `IATType` is only used as a marker for "filter me out", which betrays the purpose of the (admittedly-already-broken) `DependencyType`.

Proposal: the type of `pluginAnalyze` can change to emit `m ([ValidVSILocator], [VSIUserDep])`, where `VSIUserDep` is a new type along the lines of

```hs
data VSIUserDep
  { vsiUserDepName :: Text
  , vsiUserDepVersion :: Text
  } 
```

`ValidVSILocator`s can be later processed into `Dependency`s, and `VSIUserDep`s are marshalled into additionaldepdata

---

More generally: it's best to keep our data in well-behaved, correct-by-construction internal types for as long as we can, delaying the conversion to something much less well-behaved (`Dependency`, which has fewer constraints on its shape) until the very last moment

After something is made into a `Dependency`, we should never touch it again